src/libraries/System.Private.Xml/src/System/Xml/XPath/Internal/FunctionQuery.cs

```diff
@@ -86,17 +90,17 @@ public override object Evaluate(XPathNodeIterator nodeIterator)
             }
         }
 
-        public override XPathNavigator MatchNode(XPathNavigator navigator)
+        public override XPathNavigator? MatchNode(XPathNavigator? navigator)
         {
             if (name != "key" && prefix.Length != 0)
             {
                 throw XPathException.Create(SR.Xp_InvalidPattern);
             }
-            this.Evaluate(new XPathSingletonIterator(navigator, /*moved:*/true));
-            XPathNavigator nav = null;
+            this.Evaluate(new XPathSingletonIterator(navigator!, /*moved:*/true));
```

Comment:
consider changing base class so that navigator isn't nullable

Reply:
There are many overloads that check for null on `navigator` parameter.
See: https://github.com/dotnet/runtime/pull/39821/files#diff-731058c0b81190ef2c9ad362da38ad03R76-R81
https://github.com/dotnet/runtime/pull/39821/files#diff-ed564f040380bf361ccb08d3b51fe31eR27-R30
https://github.com/dotnet/runtime/pull/39821/files#diff-ef1adc031fa7c89f0531fdc76b3edafeR143-R146

Reply:
I'm usually conservative about those - if one overload doesn't accept it then it's easier to make it all more restrictive and fix in the future (you cannot do reverse since that would be a breaking change). Consider also just adding a null check to match what other places do

Reply:
`MatchNode` is not API (the type is internal) so we can go back to strict if in the future we can determined that the null checks are not needed.